internal/service/event_handler.go

```diff
@@ -77,8 +77,19 @@ func (h *EventHandler) HandleDeviceUpdatedEvents(ctx context.Context, resourceKi
 	// Only generate status change events when the device is not being created
 	if !created {
 		statusUpdates := common.ComputeDeviceStatusChanges(ctx, oldDevice, newDevice, orgId, h.store)
+
+		// Deduplicate DeviceDisconnected events - if multiple status fields changed to Unknown,
+		// only emit one DeviceDisconnected event
+		deviceDisconnectedEmitted := false
```

Comment:
I took a quick look and it seemed like this is the only type of event that can be duplicated across the different device statuses, but was curious about your thoughts on just generically deduplicating all events here. Not until we need it?

Reply:
In general you're right.  However, where possible, I'm emitting events as callbacks from the store layer with the resources before and after DB updates.  This should guarantee emitting events once.